index.js

```diff
@@ -1,70 +1,34 @@
-const denodeify = require('then-denodeify')
-const nodeify = require('then-nodeify')
+module.exports = function createMediator() {
+	const providers = Object.create(null)
 
-module.exports = function createAppContext() {
-	const eventsToEventListeners = Object.create(null)
-	const unsubscribeFunctions = []
+	function call(name, ...args) {
+		if (providers[name]) {
+			try {
+				const response = providers[name](...args)
 
-	function callListeners(event, ...eventArguments) {
-		const args = []
-
-		eventArguments.forEach((argument, i) => {
-			const isCallbackFunction = i === eventArguments.length - 1 && typeof argument === 'function'
-
-			if (!isCallbackFunction) {
-				args.push(argument)
-			}
-		})
-
-		return new Promise((resolve, reject) => {
-			if (eventsToEventListeners[event]) {
-				values(eventsToEventListeners[event]).forEach(listener => {
-					listener(...args).then(resolve, reject)
-				})
-			}
-		})
-	}
-	function subscribe(event, cb) {
-		const unsubscribeEvent = addListener(eventsToEventListeners, event, cb)
-		unsubscribeFunctions.push(unsubscribeEvent)
-
-		return function unsubscribe() {
-			unsubscribeEvent()
-			const index = unsubscribeFunctions.indexOf(unsubscribeEvent)
-			if (index !== -1) {
-				unsubscribeFunctions.splice(index, 1)
+				return Promise.resolve(response)
+			} catch (err) {
+				return Promise.reject(err)
```

Comment:
Mediator errors should be like SQL query syntax errors - they should cause loud obvious failures to be caught in development, and not dealt with by run-time code.

This particular error isn't technically a mediator error, it would be caused by run-time errors thrown by the provider function.

The other errors that the mediator is actually constructing itself, though - I don't think there would be any benefit to making them detectable.  I'm expecting them to bubble all the way back out to the developer.

In a perfect world that sort of error "you haven't yet set a provider for this function name" would be caught at compile-time, not run-time.